RevenueCatUI/CustomerCenter/Data/CustomerCenterData.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  File.swift
+//  
+//
+//  Created by Cesar de la Vega on 28/5/24.
+//
+
+import Foundation
+
+struct CustomerCenterData: Decodable {
+
+    let id: String
+    let paths: [HelpPath]
+    let title: LocalizedString
+    let supportEmail: String
+    let appearance: Appearance
+
+    enum HelpPathType: String, Decodable {
+        case missingPurchase = "MISSING_PURCHASE"
+        case refundRequest = "REFUND_REQUEST"
+        case changePlans = "CHANGE_PLANS"
+        case cancel = "CANCEL"
+        case unknown
+    }
+
+    struct HelpPath: Decodable {
+
+        let id: String
+        let title: LocalizedString
+        let type: HelpPathType
+        let promotionalOffer: PromotionalOffer?
+        let feedbackSurvey: FeedbackSurvey?
```

Comment:
Is it always going to be either a `promotionalOffer` OR a `feedbackSurvey`, never both? If so, is there a way to enforce that at compile time? In Kotlin I would suggest a `sealed class/interface`. Can an enum with associated values do something like that in Swift?

Reply:
Thoughts on enforcing that on compilation in Swift:

- you could have both types conform to the same protocol and then have a property of that protocol, i.e.: 
```swift
protocol Foo: Decodable {}

struct PromotionalOffer: Foo {
}

struct FeedbackSurvey: Foo {
}

...
    let offerOrSurvey: Foo

```

- alternatively, you could have them be enums with associated values, i.e.: 
```swift
enum Foo {
        case promotionalOffer(PromotionalOffer)
        case feedbackSurvey(FeedbackSurvey)
}
```

I think I'd probably go with the latter but I'm not really a Swift expert either

Reply:
done in 553edba494998359e5add7abcf013f8dbc16143d